### PR TITLE
Separated into sections and added HP Tool Configuration Generation

### PR DESCRIPTION
**Pull request by justaduck**
Separated into sections (GUI and Code) and added HP Tool Configuration Generation

### CODE REVIEW

File: scripts/python/jtprefs.py [hunk 1]
```diff
@@ -85,66 +129,117 @@ def submit_data(self):
...
 
-def create_json(username, organization, s3info, apiurl, apitoken):
-    with open(os.path.expanduser("~\\.maskgen2"), "w") as f:
-        data = {"username": username, "apitoken": apitoken, "organization": organization,
-                "s3info": s3info, "apiurl": apiurl}
-        json_data = json.dumps(data, indent=4)
+def combine_settings(mg_loader):
+    with open(hp_settings, "r") as hp:
+        hp_config = json.load(hp)
+    with open(maskgen2, "r") as jt:
```
Comment (rwgdrummer) [inline scripts/python/jtprefs.py]: The MaskgenLoader already opens the file.  This can be removed.

File: scripts/python/jtprefs.py [hunk 2]
```diff
@@ -85,66 +129,117 @@ def submit_data(self):
...
-            tkMessageBox.showerror("Invalid API Token", self.apitoken)
+    def get_recipient(self):
+        if not os.path.isfile(key):
+            return None
+
+        gpg_result = subprocess.Popen(["gpg", "--with-colons", key], stdout=subprocess.PIPE,
+                                      stderr=subprocess.PIPE).communicate()
 
-    def get_token(self, url, username, password):
+        email = gpg_result[0].split("<")[1].split(">")[0]
```
Comment (rwgdrummer) [inline scripts/python/jtprefs.py]: The UI is not guaranteed to be in [0]. 
Look for the line (list entry) that starts with 'uid'

File: scripts/python/jtprefs.py [hunk 3]
```diff
@@ -2,79 +2,123 @@
...
 import tkMessageBox
 import json
-
 import requests
+import subprocess
+from maskgen.maskgen_loader import MaskGenLoader
 
+key = os.path.join(os.path.expanduser("~"), "medifor_ingest.gpg")
+hp_settings = os.path.join(os.path.expanduser("~"), ".hpsettings")
+maskgen2 = os.path.join(os.path.expanduser("~"), ".maskgen2")
```
Comment (rwgdrummer) [inline scripts/python/jtprefs.py]: not needed.  MaskGenLoader handles this.  Remove

File: scripts/python/jtprefs.py (hunk 1 above)
Comment (rwgdrummer) [reply scripts/python/jtprefs.py]: Also.  Check to make sure the maskgen settings is 'master' so the hp settings does overload it.  Or should it be the other way around?

File: hp_tool/hp/prefs.py [hunk 4]
```diff
@@ -14,7 +14,7 @@ class SettingsManager():
     """
     def __init__(self, settingsFile=None):
         if settingsFile is None:
-            self.settingsFile = os.path.join(os.path.expanduser('~'), '.hpsettings')
+            self.settingsFile = os.path.join(os.path.expanduser('~'), '.maskgen2')
```
Comment (rwgdrummer) [inline hp_tool/hp/prefs.py]: Remove

File: hp_tool/hp/hp_data.py [hunk 5]
```diff
@@ -17,10 +17,20 @@
 import json
 import data_files
 
-exts = {'IMAGE':['.jpg', '.jpeg', '.png', '.tif', '.tiff', '.nef', '.crw', '.cr2', '.dng', '.arw', '.srf', '.raf'],
-        'VIDEO':['.avi', '.mov', '.mp4', '.mpg', '.mts', '.asf','.mxf'],
-        'AUDIO':['.wav', '.mp3', '.flac', '.webm', '.aac', '.amr', '.3ga'],
-        'MODEL': ['.3d.zip']}
+if not os.path.exists(data_files._FILETYPES):
+    exts = {'IMAGE': ['.jpg', '.jpeg', '.png', '.tif', '.tiff', '.nef', '.crw', '.cr2', '.dng', '.arw', '.srf',
```
Comment (rwgdrummer) [inline hp_tool/hp/hp_data.py]: By DEFAULT:
using maskgen.tool_set imagefiletypes,videofiletypes,audiofiletypes.
Move these file types missing from those lists into those lists.  That way we have one way of doing all. 
E.g. 3ga missing from the audiofiletypes

File: scripts/python/jtprefs.py (hunk 2 above)
Comment (justaduck) [reply scripts/python/jtprefs.py]: gpg_result[0] is stdout.  It isn't splitting at any specific line, just the < and >.  Do you still want it to split at a specific line?

File: scripts/python/jtprefs.py (hunk 1 above)
Comment (justaduck) [reply scripts/python/jtprefs.py]: Anything that could possibly collide, is passed through a renaming process.  Settings that would remain constant through the two, and both need to reference keep whatever is in maskgen.

**Pull request closed without merging**